src/main/java/Hello.java

```diff
@@ -37,19 +73,25 @@ public static void main(String... args) throws Exception {
     boolean flagValue = client.boolVariation(FEATURE_FLAG_KEY, context, false);
     showMessage("Feature flag '" + FEATURE_FLAG_KEY + "' is " + flagValue + " for this context");
 
-    // Here we request that the SDK flush pending analytic events so that you see
-    // data for the above evaluation on the dashboard immediatelynow rather than
-    // at the next automatic flush interval. You don't need to do this under normal
-    // circumstances in your own application.
-    client.flush();
+    if (flagValue) {
+      printLogo();
+    }
+
+    //If this is building for CI, we don't need to keep running the Hello App continously.
+    if(CIMode) {
+      System.exit(0);
+    }
 
     // We set up a flag change listener so you can see flag changes as you change
     // the flag rules.
     client.getFlagTracker().addFlagChangeListener(event -> {
-      showMessage("Feature flag '" + event.getKey() + "' has changed.");
```

Comment:
Instead of using the generic flag change listener, you should consider using the `addFlagValueChangeListener`. This way you can ONLY react to the flag of interest, and only when the value changes. This removes a bunch of the duplication you have in here. It also prevents you from erroneous printing a message when the flag has changed, but not in a way that affected the outcome.